espflash/src/cli/mod.rs

```diff
@@ -134,6 +138,9 @@ pub struct FlashArgs {
     /// Load the application to RAM instead of Flash
     #[arg(long)]
     pub ram: bool,
+    /// Logging format.
+    #[arg(long, short = 'f', default_value = "serial", requires = "monitor")]
+    pub log_format: LogFormat,
```

Comment:
As @jessebraham pointed, this is a major breaking change, tomorrow I will try to check if there is any workaround for this.

Reply:
This is breaking for library users because a cargo feature got removed, but other than that, do the version of the binary artifacts matter much? Am I missing some other breakage?

Is there much reason for keeping the version bump minor and spend time trying to work around the breakage? Version numbers are supposedly cheap.

Reply:
I couldn't find a way to workaround this in a non-breaking change manner. 

> This is breaking for library users because a cargo feature got removed, but other than that, do the version of the binary artifacts matter much? Am I missing some other breakage?
> 
> Is there much reason for keeping the version bump minor and spend time trying to work around the breakage? Version numbers are supposedly cheap.

I guess we should try to be SemVer compatible to avoid breaking our dependents applications, that being said we could evaluate if we should include it for a minor release anyway. Cc @esp-rs/espressif

Reply:
We decided to start working towards v3, so this PR can now be merged